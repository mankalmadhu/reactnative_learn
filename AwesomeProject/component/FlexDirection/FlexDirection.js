import React, { Component } from 'react';
import { AppRegistry, View } from 'react-native';
import styles from './FlexDirection.Styles';
import ImageHolder from '../ImageHolder/ImageHolder';

export default class FlexDirection extends Component {
  render() {
    return (
      <View style={styles.viewtop}>
        <View style={styles.viewinner1} >
          <ImageHolder pick={'https://upload.wikimedia.org/wikipedia/commons/d/de/Bananavarieties.jpg'}/>
        </View>
        <View style={styles.viewinner2} />
        <View style={styles.viewinner3} />
      </View>
    );
  }
};