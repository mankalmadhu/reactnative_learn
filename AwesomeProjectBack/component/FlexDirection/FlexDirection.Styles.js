import {StyleSheet} from 'react-native';

module.exports  = StyleSheet.create({
  viewtop: {flex: 1, flexDirection: 'row'},
  viewinner1:{width: 50, height: 50, backgroundColor: 'powderblue'},
  viewinner2:{width: 50, height: 50, backgroundColor: 'skyblue'},
  viewinner3:{width: 50, height: 50, backgroundColor: 'green'},
  
});