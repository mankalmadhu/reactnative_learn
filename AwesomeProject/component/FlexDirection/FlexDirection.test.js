import 'react-native';
import React from 'react';
import FlexDirection from './FlexDirection';
import {View, StyleSheet} from 'react-native';

import Renderer from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow';

const ArticleTitle = props => (
  <span>ArticleTitle {JSON.stringify(props)}</span>
);


jest.mock('react-native', () => {
  const React = require('React');
  class View extends React.Component {
    render() {
      return React.createElement('View' , this.props.style);
    }
  } 
     
  return ({
   View: View,
   StyleSheet: {
      create: () => ({})
    }
  });
});

test("Flex Direction Render" , () => {
   const flexDircetion = Renderer.create(
       <FlexDirection/>
   ).toJSON();

   expect(flexDircetion).toMatchSnapshot();

});

test("Flex Direction Shallow Render" , () => {

   const shallowRenderer = new ShallowRenderer();
   shallowRenderer.render(<FlexDirection/>);
   const result = shallowRenderer.getRenderOutput();
   expect(result.props.children.length).toBe(3);

});
