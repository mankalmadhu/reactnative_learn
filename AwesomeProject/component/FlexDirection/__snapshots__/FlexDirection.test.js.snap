// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flex Direction Render 1`] = `<View />`;
